code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,684 @@
+		playsound(src, 'sound/effects/internals.ogg', 100)
+		tank.forceMove(breather)
+		breather.internal = tank
+		if(breather.internals)
+			breather.internals.icon_state = "internal1"
+		valve_open = TRUE
+		update_icon()
+		return
+	if(epp_active)
+		var/response = alert(usr, "Are you sure you want to close \the [tank]'s valve? The Emergency Positive Pressure system is currently active!", "Toggle Valve", "Yes", "No")
+		if(response == "No")
+			return
+		epp_active = FALSE
+	usr.visible_message("<b>[usr]</b> closes \the [tank]'s valve.", SPAN_NOTICE("You close \the [tank]'s valve."))
+	playsound(src, 'sound/effects/internals.ogg', 100)
+	tank_off()
+	update_icon()
+
+/obj/machinery/iv_drip/verb/toggle_epp()
+	set category = "Object"
+	set name = "Toggle EPP"
+	set src in view(1)
 
-	to_chat(usr, "<span class='notice'>[attached ? attached : "No one"] is attached.</span>")
+	if(!toggle_check())
+		return
+	if(epp_active)
+		var/response = alert(usr, "Are you sure you want to turn off the Emergency Positive Pressure system? It is currently active!", "Toggle EPP", "Yes", "No")
+		if(response == "No")
+			return
+		epp_active = FALSE
+	epp = !epp
+	usr.visible_message("<b>[usr]</b> toggles \the [src]'s Emergency Positive Pressure system [epp ? "on" : "off"].", SPAN_NOTICE("You toggle \the [src]'s Emergency Positive Pressure system [epp ? "on" : "off"]."))
+	playsound(usr, 'sound/machines/click.ogg', 50)
 
-// Let's doctors set the rate of transfer. Useful if you want to set the rate at the rate of metabolisation.
-// No longer have to take someone to dialysis because they have leftover sleeptox after surgery.
 /obj/machinery/iv_drip/verb/transfer_rate()
 	set category = "Object"
 	set name = "Set Transfer Rate"
 	set src in view(1)
 
-	if (!ishuman(usr) && !issilicon(usr))
-		return
-	if (usr.stat || usr.restrained() || !Adjacent(usr))
+	if(!toggle_check())
 		return
 	set_rate:
 		var/amount = input("Set transfer rate as u/sec (between 4 and 0.001)") as num
 		if ((0.001 > amount || amount > 4) && amount != 0)
-			to_chat(usr, "<span class='warning'>Entered value must be between 0.001 and 4.</span>")
+			to_chat(usr, SPAN_WARNING("Entered value must be between 0.001 and 4."))
 			goto set_rate
 		if (transfer_amount == 0)
 			transfer_amount = REM
 			return
 		transfer_amount = amount
-		to_chat(usr, "<span class='notice'>Transfer rate set to [src.transfer_amount] u/sec</span>")
+		to_chat(usr, SPAN_NOTICE("Transfer rate set to [src.transfer_amount] u/sec"))
 
-/obj/machinery/iv_drip/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
-	if(height && istype(mover) && mover.checkpass(PASSTABLE)) //allow bullets, beams, thrown objects, rats, drones, and the like through.
-		return 1
-	return ..()
-
-/obj/machinery/iv_drip/AltClick(mob/user)
-	. = ..()
-	transfer_rate()
+/obj/machinery/iv_drip/examine(mob/user)
+	..(user)
+	if (!(user in view(2)) && user!=src.loc)
```

Comment:
```suggestion
	if (!(user in viewers(2)) && user!=src.loc)
```

Better to use `viewers` than `view` here because byond optimizes it a bit and only returns mobs from `viewers`.

Also what does the second part of this check do? What situation would cause the IV drip to be inside the mob examining it?

Reply:
I see, I'll change it shortly.

Also, I have no idea at all what the second part is about since that whole string ``if(!(user in view(2)) && user != src.loc`` is part of the original IV code which I didn't mess with. I'll clear it up though!